### DISCUSSION TITLE
Lesson 7: Using Mocks

### DISCUSSION THREAD

**Discussion by kailashchoudhary11** [smartcontractkit/full-blockchain-solidity-course-py#1830] [Q&A] [answered]
How do I get the sub_id  after deploying VRFCordinatorMock?

**Comment by Niferu** [comment, marked as answer]
Hi @kailashchoudhary11 

You can do below to get sub_id:

```python
# Creating Subscription And Getting SubscriptionId...
create_sub_tx = vrfCoordinatorV2Mock.createSubscription()
create_sub_tx.wait(1)
subId = create_sub_tx.return_value
# Second way to get subscription id
subID = create_sub_tx.events["SubscriptionCreated"]["subId"]
print(f'SubscriptionId: {subId, subID}')
```

Just note that sub_id is used in VRF_V2 not in previous 1st version. You can also take a look at my whole lottery on VRF_V2 here: https://github.com/Neftyr/Lottery-On-VRFConsumerBaseV2-Python

**Reply by kailashchoudhary11** [reply]
Thank you! It was helpful. But did you get this error? 
VRFCoordinatorV2Mock.constructor confirmed (code size to deposit exceeds maximum code size)   Block: 1   Gas used: 1000000 (100.00%)

**Reply by Niferu** [reply]
@kailashchoudhary11 

Hmm never seen such error, but maybe @cromewar have seen it?

**Reply by cromewar** [reply]
This happens when the contract is too big for the limit on Solidity, try to split the contract o smaller ones with specific functions.

**Reply by kailashchoudhary11** [reply]
@Neftyr You did not get this error while deploying the VRFCoordinatorV2Mock contract? I am deploying the same which you have added to your github repo and getting this error

**Reply by Niferu** [reply]
@kailashchoudhary11 

Hmm thats bizzare, it is working perfectly good for me without any errors. Network also does not matter as it works both on local and goerli testnet.

**Reply by kailashchoudhary11** [reply]
> @kailashchoudhary11
> 
> Hmm thats bizzare, it is working perfectly good for me without any errors. Network also does not matter as it works both on local and goerli testnet.

Yah, but it is not working for me. I tried to deploy it by creating another brownie project but still got the same error.

**Reply by Niferu** [reply]
@kailashchoudhary11 

You can link me your repo so I can take a look at this, but VRF_V2 mock contract should be easily deployable as a lot of developers using it and none has such error as I can see.

**Reply by kailashchoudhary11** [reply]
@Neftyr 
Sure here is the link of the repo
https://github.com/kailashchoudhary11/smartcontract-lottery

**Reply by cromewar** [reply]
I will give a check too

**Reply by kailashchoudhary11** [reply]
@cromewar @Neftyr  Did you'll find anything wrong in the code?

**Reply by cromewar** [reply]
Not particularly, I'll clone your repo and try to reproduce the error.

**Reply by Niferu** [reply]
@kailashchoudhary11 

Sorry I was out for 1 week vacation have you guys solved issue and figured out what was wrong?

**Reply by kailashchoudhary11** [reply]
@Neftyr yeah the issue is resolved. There was some problem with the brownie environment.
